### ISSUE TITLE
Off-by-one bug in `SingleTileMergeMethods.merge`

### ISSUE THREAD

**Issue by metasim** [locationtech/geotrellis#2339]
I've run into a problem restitching a set of tiles back into the original raster size, which I think is caused by an off-by-one bug in [`SingleTileMergeMethods.merge`](https://github.com/locationtech/geotrellis/blob/ac4a6c22de90ca53777912e028a83489253ca03a/raster/src/main/scala/geotrellis/raster/merge/SinglebandTileMergeMethods.scala#L116). 

Using the debugger, I've reduced my input and the relevant code to the following example and a test for what I'd expect the result to be:

```scala
import geotrellis.raster.{CellSize, RasterExtent}
import geotrellis.vector.Extent
val size = 64
val extent = Extent(431902.5, 4320187.5, 432862.5, 4321147.5)
val re = RasterExtent(extent, size, size)
// re: geotrellis.raster.RasterExtent = GridExtent(Extent(431902.5, 4320187.5, 432862.5, 4321147.5),15.0,15.0)
val gb = re.gridBoundsFor(extent)
// gb: geotrellis.raster.GridBounds = GridBounds(0,0,63,63)
val targetCS = CellSize(extent, gb.colMax, gb.rowMax)
// targetCS: geotrellis.raster.CellSize = CellSize(15.238095238095237,15.238095238095237)
assert(targetCS == re.cellSize, s"$targetCS != ${re.cellSize}")

java.lang.AssertionError: assertion failed: CellSize(15.238095238095237,15.238095238095237) != CellSize(15.0,15.0)
  ... 42 elided
```
Because the starting cell size and the ending cell size don't end up being the same in my specific case, I end up with at merged raster that is four times as big as it should be, with ¾ of that space being `NoData`.

As @pomadchin [states here](https://gitter.im/geotrellis/geotrellis?at=599a9e58578b44a046d08cd0), I think `CellSize(extent, gb.colMax, gb.rowMax)` should actually be `CellSize(extent, gb.width, gb.height)`.

While I don't think it's related, note that #2330 deals with `RasterExtent.gridBoundsFor`, also used in this computation.

**Comment by metasim** [comment]
While the cell size computation may be in error, I'm not convinced that it's causing my underlying problem. The cell size is only used in [specific resampling routines](https://github.com/locationtech/geotrellis/blob/master/raster/src/main/scala/geotrellis/raster/resample/Resample.scala#L108). While I think it's a bug, it's likely only apparent in very specific cases.

**Comment by metasim** [comment]
@rossbernet What do you think about this making it into 1.2?

**Comment by jamesmcclain** [comment]
@metasim It looks as though you have already done some work on this?

**Comment by metasim** [comment]
Needs a PR that changes `CellSize(extent, gb.colMax, gb.rowMax)` to `CellSize(extent, gb.width, gb.height)`. Not sure what the test should be, however.

**Comment by metasim** [comment]
Addressed by #2404
